### DISCUSSION TITLE
How do you generate rust client code from a smithy IDL file?

### DISCUSSION THREAD

**Discussion by catshow** [smithy-lang/smithy-rs#2642] [Q&A] [answered]
Is there any sample code that shows how to generate rust generated code starting with a .smithy IDL file?

**Comment by ysaito1001** [comment, marked as answer]
Hi @catshow, thank you for posting a question. You can refer to the [pokemon-service](https://github.com/awslabs/smithy-rs/tree/main/examples) example.

In the `smithy-rs/examples` directory, if you run
```
make build
```
it will generate a `pokemon-service-client` crate, which is a Rust client for the service defined by the [model](https://github.com/awslabs/smithy-rs/blob/main/codegen-server-test/python/model/pokemon.smithy).
